native/com_wolfssl_WolfSSLSession.c

```diff
@@ -1090,7 +1090,7 @@ JNIEXPORT jint JNICALL Java_com_wolfssl_WolfSSLSession_read
                            ret == WOLFJNI_IO_EVENT_FAIL) {
                     /* Java will throw SocketTimeoutException or
                      * SocketException */
-                    break;
+                    return ret;
```

Comment:
We can't return directly here since we need to drop down and call `ReleaseByteArrayElements()` below the while loop.

We can do:

```
size = ret;
break;
```

But, since that's the same as the "else" block below it, we should consolodate the two together. I recommend one "else" block that does the above, but adjust the code comment inside the else to include that Java will throw SocketTimeoutException or SocketException in the case of WOLFJNI_IO_EVENT_TIMEOUT, WOLFJNI_IO_EVENT_FD_CLOSED, WOLFJNI_IO_EVENT_ERROR, WOLFJNI_IO_EVENT_POLLHUP, and WOLFJNI_IO_EVENT_FAIL.